// @resumejs/components@0.9.3/dist/index.mjs
"use client";
import Se, { useState as gr, useEffect as ke, useMemo as hr } from "react";
import yr from "react-markdown";
import br from "remark-frontmatter";
import Er from "remark-gfm";
import Ne from "clsx";
import Rr from "rehype-raw";
import _r from "@resumejs/template-default";
import { TemplateProvider as Tr } from "@resumejs/template";
import { visit as k } from "unist-util-visit";
import kr from "yaml";
import { u as D } from "unist-builder";
import { is as F } from "unist-util-is";
var Q = { exports: {} }, $ = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ce;
function Cr() {
  if (Ce)
    return $;
  Ce = 1;
  var n = Se, i = Symbol.for("react.element"), m = Symbol.for("react.fragment"), f = Object.prototype.hasOwnProperty, b = n.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, E = { key: !0, ref: !0, __self: !0, __source: !0 };
  function R(y, l, p) {
    var d, g = {}, w = null, Y = null;
    p !== void 0 && (w = "" + p), l.key !== void 0 && (w = "" + l.key), l.ref !== void 0 && (Y = l.ref);
    for (d in l)
      f.call(l, d) && !E.hasOwnProperty(d) && (g[d] = l[d]);
    if (y && y.defaultProps)
      for (d in l = y.defaultProps, l)
        g[d] === void 0 && (g[d] = l[d]);
    return { $$typeof: i, type: y, key: w, ref: Y, props: g, _owner: b.current };
  }
  return $.Fragment = m, $.jsx = R, $.jsxs = R, $;
}
var W = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Oe;
function Or() {
  return Oe || (Oe = 1, process.env.NODE_ENV !== "production" && function() {
    var n = Se, i = Symbol.for("react.element"), m = Symbol.for("react.portal"), f = Symbol.for("react.fragment"), b = Symbol.for("react.strict_mode"), E = Symbol.for("react.profiler"), R = Symbol.for("react.provider"), y = Symbol.for("react.context"), l = Symbol.for("react.forward_ref"), p = Symbol.for("react.suspense"), d = Symbol.for("react.suspense_list"), g = Symbol.for("react.memo"), w = Symbol.for("react.lazy"), Y = Symbol.for("react.offscreen"), ee = Symbol.iterator, Fe = "@@iterator";
    function Ae(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = ee && e[ee] || e[Fe];
      return typeof r == "function" ? r : null;
    }
    var N = n.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function _(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), a = 1; a < r; a++)
          t[a - 1] = arguments[a];
        Ie("error", e, t);
      }
    }
    function Ie(e, r, t) {
      {
        var a = N.ReactDebugCurrentFrame, u = a.getStackAddendum();
        u !== "" && (r += "%s", t = t.concat([u]));
        var c = t.map(function(s) {
          return String(s);
        });
        c.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, c);
      }
    }
    var $e = !1, We = !1, Ye = !1, Me = !1, Le = !1, Ue = Symbol.for("react.module.reference");
    function Ve(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === f || e === E || Le || e === b || e === p || e === d || Me || e === Y || $e || We || Ye || typeof e == "object" && e !== null && (e.$$typeof === w || e.$$typeof === g || e.$$typeof === R || e.$$typeof === y || e.$$typeof === l || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === Ue || e.getModuleId !== void 0));
    }
    function qe(e, r, t) {
      var a = e.displayName;
      if (a)
        return a;
      var u = r.displayName || r.name || "";
      return u !== "" ? t + "(" + u + ")" : t;
    }
    function re(e) {
      return e.displayName || "Context";
    }
    function O(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && _("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case f:
          return "Fragment";
        case m:
          return "Portal";
        case E:
          return "Profiler";
        case b:
          return "StrictMode";
        case p:
          return "Suspense";
        case d:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case y:
            var r = e;
            return re(r) + ".Consumer";
          case R:
            var t = e;
            return re(t._context) + ".Provider";
          case l:
            return qe(e, e.render, "ForwardRef");
          case g:
            var a = e.displayName || null;
            return a !== null ? a : O(e.type) || "Memo";
          case w: {
            var u = e, c = u._payload, s = u._init;
            try {
              return O(s(c));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var S = Object.assign, A = 0, te, ne, ae, oe, ie, se, ue;
    function le() {
    }
    le.__reactDisabledLog = !0;
    function Be() {
      {
        if (A === 0) {
          te = console.log, ne = console.info, ae = console.warn, oe = console.error, ie = console.group, se = console.groupCollapsed, ue = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: le,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        A++;
      }
    }
    function He() {
      {
        if (A--, A === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: S({}, e, {
              value: te
            }),
            info: S({}, e, {
              value: ne
            }),
            warn: S({}, e, {
              value: ae
            }),
            error: S({}, e, {
              value: oe
            }),
            group: S({}, e, {
              value: ie
            }),
            groupCollapsed: S({}, e, {
              value: se
            }),
            groupEnd: S({}, e, {
              value: ue
            })
          });
        }
        A < 0 && _("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var B = N.ReactCurrentDispatcher, H;
    function M(e, r, t) {
      {
        if (H === void 0)
          try {
            throw Error();
          } catch (u) {
            var a = u.stack.trim().match(/\n( *(at )?)/);
            H = a && a[1] || "";
          }
        return `
` + H + e;
      }
    }
    var G = !1, L;
    {
      var Ge = typeof WeakMap == "function" ? WeakMap : Map;
      L = new Ge();
    }
    function ce(e, r) {
      if (!e || G)
        return "";
      {
        var t = L.get(e);
        if (t !== void 0)
          return t;
      }
      var a;
      G = !0;
      var u = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var c;
      c = B.current, B.current = null, Be();
      try {
        if (r) {
          var s = function() {
            throw Error();
          };
          if (Object.defineProperty(s.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(s, []);
            } catch (P) {
              a = P;
            }
            Reflect.construct(e, [], s);
          } else {
            try {
              s.call();
            } catch (P) {
              a = P;
            }
            e.call(s.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (P) {
            a = P;
          }
          e();
        }
      } catch (P) {
        if (P && a && typeof P.stack == "string") {
          for (var o = P.stack.split(`
`), T = a.stack.split(`
`), v = o.length - 1, h = T.length - 1; v >= 1 && h >= 0 && o[v] !== T[h]; )
            h--;
          for (; v >= 1 && h >= 0; v--, h--)
            if (o[v] !== T[h]) {
              if (v !== 1 || h !== 1)
                do
                  if (v--, h--, h < 0 || o[v] !== T[h]) {
                    var C = `
` + o[v].replace(" at new ", " at ");
                    return e.displayName && C.includes("<anonymous>") && (C = C.replace("<anonymous>", e.displayName)), typeof e == "function" && L.set(e, C), C;
                  }
                while (v >= 1 && h >= 0);
              break;
            }
        }
      } finally {
        G = !1, B.current = c, He(), Error.prepareStackTrace = u;
      }
      var j = e ? e.displayName || e.name : "", Te = j ? M(j) : "";
      return typeof e == "function" && L.set(e, Te), Te;
    }
    function Je(e, r, t) {
      return ce(e, !1);
    }
    function ze(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function U(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return ce(e, ze(e));
      if (typeof e == "string")
        return M(e);
      switch (e) {
        case p:
          return M("Suspense");
        case d:
          return M("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case l:
            return Je(e.render);
          case g:
            return U(e.type, r, t);
          case w: {
            var a = e, u = a._payload, c = a._init;
            try {
              return U(c(u), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var V = Object.prototype.hasOwnProperty, fe = {}, de = N.ReactDebugCurrentFrame;
    function q(e) {
      if (e) {
        var r = e._owner, t = U(e.type, e._source, r ? r.type : null);
        de.setExtraStackFrame(t);
      } else
        de.setExtraStackFrame(null);
    }
    function Ke(e, r, t, a, u) {
      {
        var c = Function.call.bind(V);
        for (var s in e)
          if (c(e, s)) {
            var o = void 0;
            try {
              if (typeof e[s] != "function") {
                var T = Error((a || "React class") + ": " + t + " type `" + s + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[s] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw T.name = "Invariant Violation", T;
              }
              o = e[s](r, s, a, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (v) {
              o = v;
            }
            o && !(o instanceof Error) && (q(u), _("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", a || "React class", t, s, typeof o), q(null)), o instanceof Error && !(o.message in fe) && (fe[o.message] = !0, q(u), _("Failed %s type: %s", t, o.message), q(null));
          }
      }
    }
    var Xe = Array.isArray;
    function J(e) {
      return Xe(e);
    }
    function Ze(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function Qe(e) {
      try {
        return me(e), !1;
      } catch {
        return !0;
      }
    }
    function me(e) {
      return "" + e;
    }
    function ve(e) {
      if (Qe(e))
        return _("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", Ze(e)), me(e);
    }
    var I = N.ReactCurrentOwner, er = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, pe, ge, z;
    z = {};
    function rr(e) {
      if (V.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function tr(e) {
      if (V.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function nr(e, r) {
      if (typeof e.ref == "string" && I.current && r && I.current.stateNode !== r) {
        var t = O(I.current.type);
        z[t] || (_('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', O(I.current.type), e.ref), z[t] = !0);
      }
    }
    function ar(e, r) {
      {
        var t = function() {
          pe || (pe = !0, _("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function or(e, r) {
      {
        var t = function() {
          ge || (ge = !0, _("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var ir = function(e, r, t, a, u, c, s) {
      var o = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: i,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: t,
        props: s,
        // Record the component responsible for creating this element.
        _owner: c
      };
      return o._store = {}, Object.defineProperty(o._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(o, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: a
      }), Object.defineProperty(o, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: u
      }), Object.freeze && (Object.freeze(o.props), Object.freeze(o)), o;
    };
    function sr(e, r, t, a, u) {
      {
        var c, s = {}, o = null, T = null;
        t !== void 0 && (ve(t), o = "" + t), tr(r) && (ve(r.key), o = "" + r.key), rr(r) && (T = r.ref, nr(r, u));
        for (c in r)
          V.call(r, c) && !er.hasOwnProperty(c) && (s[c] = r[c]);
        if (e && e.defaultProps) {
          var v = e.defaultProps;
          for (c in v)
            s[c] === void 0 && (s[c] = v[c]);
        }
        if (o || T) {
          var h = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          o && ar(s, h), T && or(s, h);
        }
        return ir(e, o, T, u, a, I.current, s);
      }
    }
    var K = N.ReactCurrentOwner, he = N.ReactDebugCurrentFrame;
    function x(e) {
      if (e) {
        var r = e._owner, t = U(e.type, e._source, r ? r.type : null);
        he.setExtraStackFrame(t);
      } else
        he.setExtraStackFrame(null);
    }
    var X;
    X = !1;
    function Z(e) {
      return typeof e == "object" && e !== null && e.$$typeof === i;
    }
    function ye() {
      {
        if (K.current) {
          var e = O(K.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function ur(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var be = {};
    function lr(e) {
      {
        var r = ye();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function Ee(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = lr(r);
        if (be[t])
          return;
        be[t] = !0;
        var a = "";
        e && e._owner && e._owner !== K.current && (a = " It was passed a child from " + O(e._owner.type) + "."), x(e), _('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, a), x(null);
      }
    }
    function Re(e, r) {
      {
        if (typeof e != "object")
          return;
        if (J(e))
          for (var t = 0; t < e.length; t++) {
            var a = e[t];
            Z(a) && Ee(a, r);
          }
        else if (Z(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var u = Ae(e);
          if (typeof u == "function" && u !== e.entries)
            for (var c = u.call(e), s; !(s = c.next()).done; )
              Z(s.value) && Ee(s.value, r);
        }
      }
    }
    function cr(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === l || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === g))
          t = r.propTypes;
        else
          return;
        if (t) {
          var a = O(r);
          Ke(t, e.props, "prop", a, e);
        } else if (r.PropTypes !== void 0 && !X) {
          X = !0;
          var u = O(r);
          _("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", u || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && _("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function fr(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var a = r[t];
          if (a !== "children" && a !== "key") {
            x(e), _("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", a), x(null);
            break;
          }
        }
        e.ref !== null && (x(e), _("Invalid attribute `ref` supplied to `React.Fragment`."), x(null));
      }
    }
    function _e(e, r, t, a, u, c) {
      {
        var s = Ve(e);
        if (!s) {
          var o = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (o += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var T = ur(u);
          T ? o += T : o += ye();
          var v;
          e === null ? v = "null" : J(e) ? v = "array" : e !== void 0 && e.$$typeof === i ? (v = "<" + (O(e.type) || "Unknown") + " />", o = " Did you accidentally export a JSX literal instead of a component?") : v = typeof e, _("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", v, o);
        }
        var h = sr(e, r, t, u, c);
        if (h == null)
          return h;
        if (s) {
          var C = r.children;
          if (C !== void 0)
            if (a)
              if (J(C)) {
                for (var j = 0; j < C.length; j++)
                  Re(C[j], e);
                Object.freeze && Object.freeze(C);
              } else
                _("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Re(C, e);
        }
        return e === f ? fr(h) : cr(h), h;
      }
    }
    function dr(e, r, t) {
      return _e(e, r, t, !0);
    }
    function mr(e, r, t) {
      return _e(e, r, t, !1);
    }
    var vr = mr, pr = dr;
    W.Fragment = f, W.jsx = vr, W.jsxs = pr;
  }()), W;
}
process.env.NODE_ENV === "production" ? Q.exports = Cr() : Q.exports = Or();
var Pe = Q.exports;
const Pr = function() {
  return (n, i) => {
    k(
      n,
      {
        type: "yaml"
      },
      (m) => {
        i.data.meta = kr.parse(m.value);
      }
    );
  };
}, xe = ["h1", "h2", "h3"];
function je(n, i) {
  return F(n, { type: "element" }) && i ? [i].flat().includes(n == null ? void 0 : n.tagName) : xe.includes(n == null ? void 0 : n.tagName);
}
const wr = (n) => F(n, { type: "element", tagName: "p" });
function Sr(n) {
  return F(n, { type: "element", tagName: "table" }) && xe.includes(n.tagName);
}
function Nr(n) {
  return F(n, { type: "element", tagName: "ul" });
}
function xr(n) {
  return F(n, { type: "element", tagName: "blockquote" });
}
function De(n) {
  return F(n, { type: "element", tagName: "img" });
}
const jr = function() {
  return (n) => {
    k(
      n,
      { type: "element", tagName: "table" },
      (i, m, f) => {
        if (i.children.length < 2)
          return;
        const b = f.children[m - 2];
        if (!je(b))
          return;
        const E = [];
        k(i, { tagName: "th" }, (y) => {
          y.tagName = "card-item-label", E.push(y);
        });
        const R = [];
        k(
          // tbody
          i.children[1],
          { tagName: "tr" },
          (y) => {
            let l = 0;
            const p = [];
            k(y, { tagName: "td" }, (d) => {
              const g = E[l];
              g && (d.tagName = "card-item-value", p[l++] = { label: g, value: d });
            }), R.push({
              type: "element",
              tagName: "card",
              properties: {},
              position: i.position,
              children: p.map((d, g) => D(
                "element",
                {
                  tagName: "card-item",
                  properties: {
                    index: g
                  }
                },
                [d.label, D("text", d.label ? "：" : ""), d.value]
              ))
            });
          }
        ), f == null || f.children.splice(m, 1, ...R);
      }
    );
  };
}, Dr = function() {
  return (n) => {
    k(
      n,
      {
        type: "element",
        tagName: "p"
      },
      (i, m, f) => {
        if (De(i.children[0]))
          return;
        const b = f.children[m - 2];
        (Sr(b) || je(b)) && (i.tagName = "description");
      }
    );
  };
}, Fr = function() {
  return (n) => {
    k(
      n,
      {
        type: "element",
        tagName: "h1"
      },
      (i, m, f) => {
        const b = f.children.length, E = [];
        let R;
        const y = (l) => {
          f.children[l] = D("text", "");
        };
        for (let l = m + 2; l < b; l += 2) {
          const p = f.children[l];
          if (xr(p)) {
            y(l);
            continue;
          }
          if (wr(p) && De(p.children[0])) {
            R = p.children[0], y(l);
            continue;
          }
          if (!Nr(p))
            break;
          p.tagName = "header-row", E.push(p), k(p, { tagName: "li" }, (d) => {
            d.tagName = "header-col";
          }), y(l);
        }
        R && (R.tagName = "header-avatar", E.unshift(R)), i.tagName = "header-name", E.unshift(i), f.children.splice(m, 1, {
          type: "element",
          tagName: "header",
          properties: {},
          children: E
        });
      }
    );
  };
}, Ar = function() {
  return (n) => ({
    type: "root",
    children: [
      D("element", { tagName: "container" }, [
        D("element", { tagName: "content" }, n.children)
      ])
    ]
  });
}, Ir = function() {
  return (n) => {
    k(n, { type: "element" }, (i) => {
      var f;
      const m = (f = i.properties) == null ? void 0 : f.className;
      i.properties = Object.assign(
        {},
        i.properties,
        {
          className: Ne(`r-${i.tagName}`, m)
        }
      );
    });
  };
}, $r = function() {
  return (n) => {
    k(
      n,
      {
        type: "element",
        tagName: "ul"
      },
      (i) => {
        var f;
        const m = (f = i.properties) == null ? void 0 : f.className;
        !Array.isArray(m) || !m.includes("contains-task-list") || (i.tagName = "task", i.properties.className = [], k(
          i,
          {
            type: "element",
            tagName: "li"
          },
          (b) => {
            b.tagName = "task-item", b.properties.className = [], k(
              b,
              {
                type: "element",
                tagName: "input"
              },
              (E) => {
                E.properties.type === "checkbox" && (E.tagName = "task-item-checkbox");
              }
            );
          }
        ));
      }
    );
  };
}, Wr = function() {
  return (n, i) => {
    k(
      n,
      {
        type: "element",
        tagName: "content"
      },
      (m) => m.children.unshift(
        D(
          "element",
          {
            properties: {},
            tagName: "toolbox",
            data: i.data.meta
          },
          []
        )
      )
    );
  };
}, we = "RESUME_THEME";
function Xr(n) {
  const {
    children: i,
    rehypePlugins: m = [],
    remarkPlugins: f = [],
    remarkRehypeOptions: b,
    templateContextProps: E
  } = n, [R, y] = gr(!1);
  ke(() => {
    y(localStorage.getItem(we) === "dark");
  }, []), ke(() => {
    var g;
    (g = n.onDarkClass) == null || g.call(n, "dark", R ? "add" : "remove");
  }, [R, n.onDarkClass]);
  const l = () => {
    y((g) => (localStorage.setItem(we, g ? "light" : "dark"), !g));
  }, p = () => {
    window.print();
  }, d = hr(() => n.components === !1 ? {} : {
    ..._r,
    ...n.components
  }, [n.components]);
  return /* @__PURE__ */ Pe.jsx(
    Tr,
    {
      value: {
        toggleTheme: l,
        print: p,
        ...E
      },
      children: /* @__PURE__ */ Pe.jsx(
        yr,
        {
          className: Ne(
            {
              dark: !n.onDarkClass && R
            },
            "r-resume",
            n.className
          ),
          components: d,
          remarkPlugins: [...f, br, Pr, Er],
          rehypePlugins: [
            ...m,
            $r,
            Rr,
            jr,
            Fr,
            Dr,
            Ar,
            Wr,
            Ir
          ],
          remarkRehypeOptions: {
            allowDangerousHtml: !0,
            ...b
          },
          children: i
        }
      )
    }
  );
}
export {
  Xr as Resume
};
