// @resumejs/template-default@0.9.3/dist/index.mjs
import "./style.css";
import Te from "react";
import l from "clsx";
import ur from "omit.js";
import { useTemplate as cr } from "@resumejs/template";
var Q = { exports: {} }, I = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var je;
function fr() {
  if (je)
    return I;
  je = 1;
  var t = Te, a = Symbol.for("react.element"), y = Symbol.for("react.fragment"), j = Object.prototype.hasOwnProperty, O = t.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, W = { key: !0, ref: !0, __self: !0, __source: !0 };
  function D(k, p, C) {
    var g, E = {}, T = null, Y = null;
    C !== void 0 && (T = "" + C), p.key !== void 0 && (T = "" + p.key), p.ref !== void 0 && (Y = p.ref);
    for (g in p)
      j.call(p, g) && !W.hasOwnProperty(g) && (E[g] = p[g]);
    if (k && k.defaultProps)
      for (g in p = k.defaultProps, p)
        E[g] === void 0 && (E[g] = p[g]);
    return { $$typeof: a, type: k, key: T, ref: Y, props: E, _owner: O.current };
  }
  return I.Fragment = y, I.jsx = D, I.jsxs = D, I;
}
var $ = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ke;
function dr() {
  return ke || (ke = 1, process.env.NODE_ENV !== "production" && function() {
    var t = Te, a = Symbol.for("react.element"), y = Symbol.for("react.portal"), j = Symbol.for("react.fragment"), O = Symbol.for("react.strict_mode"), W = Symbol.for("react.profiler"), D = Symbol.for("react.provider"), k = Symbol.for("react.context"), p = Symbol.for("react.forward_ref"), C = Symbol.for("react.suspense"), g = Symbol.for("react.suspense_list"), E = Symbol.for("react.memo"), T = Symbol.for("react.lazy"), Y = Symbol.for("react.offscreen"), ee = Symbol.iterator, we = "@@iterator";
    function Oe(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = ee && e[ee] || e[we];
      return typeof r == "function" ? r : null;
    }
    var P = t.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function b(e) {
      {
        for (var r = arguments.length, n = new Array(r > 1 ? r - 1 : 0), o = 1; o < r; o++)
          n[o - 1] = arguments[o];
        Ce("error", e, n);
      }
    }
    function Ce(e, r, n) {
      {
        var o = P.ReactDebugCurrentFrame, c = o.getStackAddendum();
        c !== "" && (r += "%s", n = n.concat([c]));
        var d = n.map(function(u) {
          return String(u);
        });
        d.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, d);
      }
    }
    var Pe = !1, Se = !1, Ne = !1, De = !1, Fe = !1, Ae = Symbol.for("react.module.reference");
    function Ie(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === j || e === W || Fe || e === O || e === C || e === g || De || e === Y || Pe || Se || Ne || typeof e == "object" && e !== null && (e.$$typeof === T || e.$$typeof === E || e.$$typeof === D || e.$$typeof === k || e.$$typeof === p || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === Ae || e.getModuleId !== void 0));
    }
    function $e(e, r, n) {
      var o = e.displayName;
      if (o)
        return o;
      var c = r.displayName || r.name || "";
      return c !== "" ? n + "(" + c + ")" : n;
    }
    function re(e) {
      return e.displayName || "Context";
    }
    function _(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && b("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case j:
          return "Fragment";
        case y:
          return "Portal";
        case W:
          return "Profiler";
        case O:
          return "StrictMode";
        case C:
          return "Suspense";
        case g:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case k:
            var r = e;
            return re(r) + ".Consumer";
          case D:
            var n = e;
            return re(n._context) + ".Provider";
          case p:
            return $e(e, e.render, "ForwardRef");
          case E:
            var o = e.displayName || null;
            return o !== null ? o : _(e.type) || "Memo";
          case T: {
            var c = e, d = c._payload, u = c._init;
            try {
              return _(u(d));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var w = Object.assign, F = 0, te, ae, ne, oe, ie, se, le;
    function ue() {
    }
    ue.__reactDisabledLog = !0;
    function We() {
      {
        if (F === 0) {
          te = console.log, ae = console.info, ne = console.warn, oe = console.error, ie = console.group, se = console.groupCollapsed, le = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: ue,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        F++;
      }
    }
    function Ye() {
      {
        if (F--, F === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: w({}, e, {
              value: te
            }),
            info: w({}, e, {
              value: ae
            }),
            warn: w({}, e, {
              value: ne
            }),
            error: w({}, e, {
              value: oe
            }),
            group: w({}, e, {
              value: ie
            }),
            groupCollapsed: w({}, e, {
              value: se
            }),
            groupEnd: w({}, e, {
              value: le
            })
          });
        }
        F < 0 && b("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var q = P.ReactCurrentDispatcher, G;
    function L(e, r, n) {
      {
        if (G === void 0)
          try {
            throw Error();
          } catch (c) {
            var o = c.stack.trim().match(/\n( *(at )?)/);
            G = o && o[1] || "";
          }
        return `
` + G + e;
      }
    }
    var J = !1, V;
    {
      var Le = typeof WeakMap == "function" ? WeakMap : Map;
      V = new Le();
    }
    function ce(e, r) {
      if (!e || J)
        return "";
      {
        var n = V.get(e);
        if (n !== void 0)
          return n;
      }
      var o;
      J = !0;
      var c = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var d;
      d = q.current, q.current = null, We();
      try {
        if (r) {
          var u = function() {
            throw Error();
          };
          if (Object.defineProperty(u.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(u, []);
            } catch (R) {
              o = R;
            }
            Reflect.construct(e, [], u);
          } else {
            try {
              u.call();
            } catch (R) {
              o = R;
            }
            e.call(u.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (R) {
            o = R;
          }
          e();
        }
      } catch (R) {
        if (R && o && typeof R.stack == "string") {
          for (var s = R.stack.split(`
`), x = o.stack.split(`
`), v = s.length - 1, m = x.length - 1; v >= 1 && m >= 0 && s[v] !== x[m]; )
            m--;
          for (; v >= 1 && m >= 0; v--, m--)
            if (s[v] !== x[m]) {
              if (v !== 1 || m !== 1)
                do
                  if (v--, m--, m < 0 || s[v] !== x[m]) {
                    var h = `
` + s[v].replace(" at new ", " at ");
                    return e.displayName && h.includes("<anonymous>") && (h = h.replace("<anonymous>", e.displayName)), typeof e == "function" && V.set(e, h), h;
                  }
                while (v >= 1 && m >= 0);
              break;
            }
        }
      } finally {
        J = !1, q.current = d, Ye(), Error.prepareStackTrace = c;
      }
      var N = e ? e.displayName || e.name : "", Re = N ? L(N) : "";
      return typeof e == "function" && V.set(e, Re), Re;
    }
    function Ve(e, r, n) {
      return ce(e, !1);
    }
    function Me(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function M(e, r, n) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return ce(e, Me(e));
      if (typeof e == "string")
        return L(e);
      switch (e) {
        case C:
          return L("Suspense");
        case g:
          return L("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case p:
            return Ve(e.render);
          case E:
            return M(e.type, r, n);
          case T: {
            var o = e, c = o._payload, d = o._init;
            try {
              return M(d(c), r, n);
            } catch {
            }
          }
        }
      return "";
    }
    var U = Object.prototype.hasOwnProperty, fe = {}, de = P.ReactDebugCurrentFrame;
    function B(e) {
      if (e) {
        var r = e._owner, n = M(e.type, e._source, r ? r.type : null);
        de.setExtraStackFrame(n);
      } else
        de.setExtraStackFrame(null);
    }
    function Ue(e, r, n, o, c) {
      {
        var d = Function.call.bind(U);
        for (var u in e)
          if (d(e, u)) {
            var s = void 0;
            try {
              if (typeof e[u] != "function") {
                var x = Error((o || "React class") + ": " + n + " type `" + u + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[u] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw x.name = "Invariant Violation", x;
              }
              s = e[u](r, u, o, n, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (v) {
              s = v;
            }
            s && !(s instanceof Error) && (B(c), b("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", o || "React class", n, u, typeof s), B(null)), s instanceof Error && !(s.message in fe) && (fe[s.message] = !0, B(c), b("Failed %s type: %s", n, s.message), B(null));
          }
      }
    }
    var Be = Array.isArray;
    function z(e) {
      return Be(e);
    }
    function qe(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, n = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return n;
      }
    }
    function Ge(e) {
      try {
        return ve(e), !1;
      } catch {
        return !0;
      }
    }
    function ve(e) {
      return "" + e;
    }
    function me(e) {
      if (Ge(e))
        return b("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", qe(e)), ve(e);
    }
    var A = P.ReactCurrentOwner, Je = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, pe, be, K;
    K = {};
    function ze(e) {
      if (U.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function Ke(e) {
      if (U.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function He(e, r) {
      if (typeof e.ref == "string" && A.current && r && A.current.stateNode !== r) {
        var n = _(A.current.type);
        K[n] || (b('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', _(A.current.type), e.ref), K[n] = !0);
      }
    }
    function Xe(e, r) {
      {
        var n = function() {
          pe || (pe = !0, b("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        n.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: n,
          configurable: !0
        });
      }
    }
    function Ze(e, r) {
      {
        var n = function() {
          be || (be = !0, b("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        n.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: n,
          configurable: !0
        });
      }
    }
    var Qe = function(e, r, n, o, c, d, u) {
      var s = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: a,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: n,
        props: u,
        // Record the component responsible for creating this element.
        _owner: d
      };
      return s._store = {}, Object.defineProperty(s._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(s, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: o
      }), Object.defineProperty(s, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: c
      }), Object.freeze && (Object.freeze(s.props), Object.freeze(s)), s;
    };
    function er(e, r, n, o, c) {
      {
        var d, u = {}, s = null, x = null;
        n !== void 0 && (me(n), s = "" + n), Ke(r) && (me(r.key), s = "" + r.key), ze(r) && (x = r.ref, He(r, c));
        for (d in r)
          U.call(r, d) && !Je.hasOwnProperty(d) && (u[d] = r[d]);
        if (e && e.defaultProps) {
          var v = e.defaultProps;
          for (d in v)
            u[d] === void 0 && (u[d] = v[d]);
        }
        if (s || x) {
          var m = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          s && Xe(u, m), x && Ze(u, m);
        }
        return Qe(e, s, x, c, o, A.current, u);
      }
    }
    var H = P.ReactCurrentOwner, xe = P.ReactDebugCurrentFrame;
    function S(e) {
      if (e) {
        var r = e._owner, n = M(e.type, e._source, r ? r.type : null);
        xe.setExtraStackFrame(n);
      } else
        xe.setExtraStackFrame(null);
    }
    var X;
    X = !1;
    function Z(e) {
      return typeof e == "object" && e !== null && e.$$typeof === a;
    }
    function ge() {
      {
        if (H.current) {
          var e = _(H.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function rr(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), n = e.lineNumber;
          return `

Check your code at ` + r + ":" + n + ".";
        }
        return "";
      }
    }
    var he = {};
    function tr(e) {
      {
        var r = ge();
        if (!r) {
          var n = typeof e == "string" ? e : e.displayName || e.name;
          n && (r = `

Check the top-level render call using <` + n + ">.");
        }
        return r;
      }
    }
    function ye(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var n = tr(r);
        if (he[n])
          return;
        he[n] = !0;
        var o = "";
        e && e._owner && e._owner !== H.current && (o = " It was passed a child from " + _(e._owner.type) + "."), S(e), b('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', n, o), S(null);
      }
    }
    function Ee(e, r) {
      {
        if (typeof e != "object")
          return;
        if (z(e))
          for (var n = 0; n < e.length; n++) {
            var o = e[n];
            Z(o) && ye(o, r);
          }
        else if (Z(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var c = Oe(e);
          if (typeof c == "function" && c !== e.entries)
            for (var d = c.call(e), u; !(u = d.next()).done; )
              Z(u.value) && ye(u.value, r);
        }
      }
    }
    function ar(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var n;
        if (typeof r == "function")
          n = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === p || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === E))
          n = r.propTypes;
        else
          return;
        if (n) {
          var o = _(r);
          Ue(n, e.props, "prop", o, e);
        } else if (r.PropTypes !== void 0 && !X) {
          X = !0;
          var c = _(r);
          b("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", c || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && b("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function nr(e) {
      {
        for (var r = Object.keys(e.props), n = 0; n < r.length; n++) {
          var o = r[n];
          if (o !== "children" && o !== "key") {
            S(e), b("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", o), S(null);
            break;
          }
        }
        e.ref !== null && (S(e), b("Invalid attribute `ref` supplied to `React.Fragment`."), S(null));
      }
    }
    function _e(e, r, n, o, c, d) {
      {
        var u = Ie(e);
        if (!u) {
          var s = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (s += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var x = rr(c);
          x ? s += x : s += ge();
          var v;
          e === null ? v = "null" : z(e) ? v = "array" : e !== void 0 && e.$$typeof === a ? (v = "<" + (_(e.type) || "Unknown") + " />", s = " Did you accidentally export a JSX literal instead of a component?") : v = typeof e, b("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", v, s);
        }
        var m = er(e, r, n, c, d);
        if (m == null)
          return m;
        if (u) {
          var h = r.children;
          if (h !== void 0)
            if (o)
              if (z(h)) {
                for (var N = 0; N < h.length; N++)
                  Ee(h[N], e);
                Object.freeze && Object.freeze(h);
              } else
                b("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Ee(h, e);
        }
        return e === j ? nr(m) : ar(m), m;
      }
    }
    function or(e, r, n) {
      return _e(e, r, n, !0);
    }
    function ir(e, r, n) {
      return _e(e, r, n, !1);
    }
    var sr = ir, lr = or;
    $.Fragment = j, $.jsx = sr, $.jsxs = lr;
  }()), $;
}
process.env.NODE_ENV === "production" ? Q.exports = fr() : Q.exports = dr();
var i = Q.exports;
function vr({ github: t }) {
  return t ? /* @__PURE__ */ i.jsx(
    "a",
    {
      "aria-label": "项目地址",
      title: "github",
      href: t,
      className: l(
        "r-carbon-logo-github text-lg hover:text-gray transition-colors",
        "dark:text-light"
      )
    }
  ) : null;
}
function mr({ print: t, onClick: a }) {
  return t ? /* @__PURE__ */ i.jsx(
    "button",
    {
      title: "打印",
      onClick: a,
      className: l(
        "r-carbon-printer text-lg hover:text-gray transition-colors",
        "dark:text-light"
      )
    }
  ) : null;
}
function pr({ theme: t, onClick: a }) {
  return t ? /* @__PURE__ */ i.jsx(
    "button",
    {
      title: "主题",
      onClick: a,
      className: l(
        "r-carbon-sun text-lg hover:text-gray transition-colors",
        "dark:r-carbon-moon dark:text-light"
      )
    }
  ) : null;
}
const br = ({ meta: t, ...a }) => {
  const { toggleTheme: y, print: j, extraToolboxButton: O } = cr();
  return /* @__PURE__ */ i.jsxs(
    "div",
    {
      ...a,
      className: l(
        "absolute right-4 top-2 flex gap-x-1.5 print:hidden",
        a.className
      ),
      children: [
        O,
        /* @__PURE__ */ i.jsx(vr, { github: t.github }),
        /* @__PURE__ */ i.jsx(
          mr,
          {
            onClick: j,
            print: t.print
          }
        ),
        /* @__PURE__ */ i.jsx(
          pr,
          {
            onClick: () => y(),
            theme: t.theme
          }
        )
      ]
    }
  );
};
function f(t) {
  return ur(t, ["node", "ordered", "inline"]);
}
const _r = {
  container: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "div",
    {
      className: l(
        "font-sans text-base text-dark-400 bg-white",
        "dark:text-light-700 dark:bg-dark-400",
        t
      ),
      ...f(a)
    }
  ),
  content: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "div",
    {
      className: l(
        "border print:border-none border-light-700 md:w-full md:max-w-screen-md xs:w-full mx-auto p-y-5 sm:p-x-10 p-x-4",
        "dark:border-dark-100",
        "relative",
        t
      ),
      ...f(a)
    }
  ),
  h1: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "h1",
    {
      className: l("m-t-2 m-b-1.5 text-2xl font-semibold", t),
      ...f(a)
    }
  ),
  h2: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "h2",
    {
      className: l(
        "m-t-3 m-b-1 text-xl font-semibold p-y-1 b-b border-dark",
        "dark:border-dark-100",
        t
      ),
      ...f(a)
    }
  ),
  h3: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "h3",
    {
      className: l("m-t-2 m-b-1 p-y-0.5 text-base font-semibold", t),
      ...f(a)
    }
  ),
  ul: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "ul",
    {
      className: l(
        "m-b-2 text-0.9rem p-l-5 list-disc",
        "dark:text-gray-300",
        t
      ),
      ...f(a)
    }
  ),
  ol: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "ol",
    {
      className: l("m-b-2 text-0.9rem p-l-5 list-decimal", t),
      ...f(a)
    }
  ),
  li: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "li",
    {
      className: l("p-y-0.5", t),
      ...f(a)
    }
  ),
  a: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "a",
    {
      className: l(
        "text-blue-600 underline",
        "dark:text-blue-400",
        t
      ),
      ...f(a)
    }
  ),
  p: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "p",
    {
      className: l("", t),
      ...f(a)
    }
  ),
  img: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "img",
    {
      className: l("inline", t),
      ...f(a)
    }
  ),
  strong: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "strong",
    {
      className: l("font-semibold", t),
      ...f(a)
    }
  ),
  card: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "div",
    {
      className: l("p-y-0.5 grid grid-cols-2 gap-y-0.5", t),
      ...f(a)
    }
  ),
  "card-item": ({ className: t, index: a, ...y }) => /* @__PURE__ */ i.jsx(
    "div",
    {
      ...f(y),
      className: l(
        "text-sm",
        { "justify-self-end": a % 2 === 1 },
        t
      )
    }
  ),
  "card-item-label": ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "span",
    {
      ...f(a),
      className: l("text-sm font-medium", t)
    }
  ),
  "card-item-value": ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "span",
    {
      className: l(
        "text-blue-gray-600",
        "dark:text-blue-gray-300",
        t
      ),
      ...f(a)
    }
  ),
  description: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "p",
    {
      ...f(a),
      className: l("text-sm indent", t)
    }
  ),
  header: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "header",
    {
      ...f(a),
      className: l("flex flex-col items-center p-4", t)
    }
  ),
  "header-content": ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "div",
    {
      ...f(a),
      className: l("flex flex-col items-center", t)
    }
  ),
  "header-name": ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "h1",
    {
      ...f(a),
      className: l("font-bold text-3xl", t)
    }
  ),
  "header-avatar": ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "img",
    {
      ...f(a),
      className: l("w-40 text-3xl", t)
    }
  ),
  "header-row": ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "ul",
    {
      ...f(a),
      className: l("list-none flex flex-wrap", t)
    }
  ),
  "header-col": ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "li",
    {
      className: l(
        "sibling:before:content-[quoted:|] sibling:before:p-x-2 sibling:before:text-gray-400",
        t
      ),
      ...f(a)
    }
  ),
  table: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "table",
    {
      className: l(
        "border border-dark text-sm",
        "dark:border-dark-100",
        t
      ),
      ...f(a)
    }
  ),
  th: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "th",
    {
      className: l(
        "border p-2 border-dark",
        "dark:border-dark-100",
        t
      ),
      ...a
    }
  ),
  td: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "td",
    {
      className: l(
        "border p-2 border-dark",
        "dark:border-dark-100",
        t
      ),
      ...a
    }
  ),
  blockquote: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "blockquote",
    {
      className: l(
        "border-l-0.3rem p-l-2 border-gray-200 m-b-4 text-gray-700",
        "dark:text-gray-400",
        t
      ),
      ...f(a)
    }
  ),
  code: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "code",
    {
      className: l(
        "p-x-1.5 p-y-0.5 bg-neutral-100 rounded-md text-xs",
        "dark:bg-neutral-500",
        t
      ),
      ...f(a)
    }
  ),
  task: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "ul",
    {
      className: l("m-b-2 text-0.9rem p-y-1", t),
      ...f(a)
    }
  ),
  "task-item": ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "li",
    {
      className: l("p-x-5", t),
      ...f(a)
    }
  ),
  "task-item-checkbox": ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "input",
    {
      className: l("-m-l-5 m-r-1 m-b-0.5 align-middle", t),
      ...f(a)
    }
  ),
  pre: ({ className: t, ...a }) => /* @__PURE__ */ i.jsx(
    "pre",
    {
      className: l("bg-light p-x-5 p-y-3", "dark:bg-dark-300", t),
      ...f(a)
    }
  ),
  toolbox: ({ node: t, ...a }) => {
    const y = t.data || {};
    return /* @__PURE__ */ i.jsx(
      br,
      {
        ...f(a),
        meta: y
      }
    );
  }
};
export {
  _r as default
};
